modules/alloy/src/main/ts/ephox/alloy/api/component/GuiFactory.ts

```diff
@@ -12,6 +12,11 @@ import * as Component from './Component';
 import { AlloyComponent } from './ComponentApi';
 import { AlloySpec, PremadeSpec, SimpleOrSketchSpec, SketchSpec } from './SpecTypes';
 
+export interface ExternalElement {
+  uid?: string;
```

Comment:
I realise it's not relevant to this PR, but this is missing a `readonly`. I'm starting to notice it.

Reply:
Yeah there's a lot in alloy and other places missing `readonly` 😞

Reply:
Made this one readonly at least for now 🤷